tests/Unit/PointwiseFunctions/GeneralRelativity/Test_GeodesicAcceleration.cpp

```diff
@@ -0,0 +1,178 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <random>
+
+#include "DataStructures/BoostMultiArray.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "Helpers/PointwiseFunctions/GeneralRelativity/TestHelpers.hpp"
+#include "NumericalAlgorithms/OdeIntegration/OdeIntegration.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GeodesicAcceleration.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpacetimeMetric.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/EqualWithinRoundoff.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+namespace {
+
+void test_circular_orbit_kerr_schild() {
+  MAKE_GENERATOR(gen);
+  const double mass = 3.2;
+  const std::array<double, 3> spin{{0., 0., 0.}};
+  const std::array<double, 3> center{{0., 0., 0.}};
+  gr::Solutions::KerrSchild kerr_schild(mass, spin, center);
+  std::uniform_real_distribution<> radius_dist(7., 100.);
+  const double radius = radius_dist(gen);
+  CAPTURE(radius);
+
+  const double angular_velocity = sqrt(mass) * 1. / (radius * sqrt(radius));
+  std::uniform_real_distribution<> time_dist(0., 100.);
+  const double time = time_dist(gen);
+  const double angle = angular_velocity * time;
+  const tnsr::I<double, 3> position{
+      {radius * cos(angle), radius * sin(angle), 0.}};
+  const tnsr::I<double, 3> velocity{{-radius * angular_velocity * sin(angle),
+                                     radius * angular_velocity * cos(angle),
+                                     0.}};
+  const tnsr::I<double, 3> expected_acceleration{
+      {-radius * square(angular_velocity) * cos(angle),
+       -radius * square(angular_velocity) * sin(angle), 0.}};
+
+  const auto christoffel =
+      get<gr::Tags::SpacetimeChristoffelSecondKind<double, 3, Frame::Inertial>>(
+          kerr_schild.variables(
+              position, 0.,
+              tmpl::list<gr::Tags::SpacetimeChristoffelSecondKind<
+                  double, 3, Frame::Inertial>>{}));
+
+  const auto geodesic_acc = gr::geodesic_acceleration(velocity, christoffel);
+  CHECK_ITERABLE_APPROX(geodesic_acc, expected_acceleration);
+}
+
+std::array<tnsr::I<double, 3>, 2> state_to_tnsr(
+    const std::array<double, 6>& state) {
+  tnsr::I<double, 3> pos_tensor{{state[0], state[1], state[2]}};
+  tnsr::I<double, 3> vel_tensor{{state[3], state[4], state[5]}};
+  return {std::move(pos_tensor), std::move(vel_tensor)};
+}
+
+struct BoostGeodesicIntegrator {
```

Comment:
Add a note that these next two structs must conform to Boost standards to be used in `odeint`?